src/lib/simulation/simulation.ts

```diff
@@ -100,21 +100,24 @@ const dataForMintInstructions = (mintInstructions: EVMMintInstructions) => {
   const abi = mintInstructions.abi;
   const iface = new Utils.Interface(abi);
 
-  const paramsArray = prepareContractParams(mintInstructions.contractParams);
+  const paramsArray = prepareContractParams(mintInstructions);
+  console.log(`Params: ${JSON.stringify(paramsArray)}`);
   const data = iface.encodeFunctionData(mintInstructions.contractMethod, paramsArray);
 
   return data;
 };
 
-export const prepareContractParams = (template: string): any[] => {
+export const prepareContractParams = (mintInstructions: EVMMintInstructions): any[] => {
   const regex = /{{(\w+)}}|tokenId|address|encodedAddress/g;
 
-  const replacedTemplate = template.replace(regex, (match) => {
+  const replacedTemplate = mintInstructions.contractParams.replace(regex, (match) => {
     switch (match) {
       case 'address':
         return `"${TEST_RECIPIENT}"`;
       case 'encodedAddress':
         return `"${encodeAddress(TEST_RECIPIENT)}"`;
+      case 'tokenId':
+        return `"${mintInstructions.tokenId || '1'}"`;
```

Comment:
are you sure you want `|| '1'` here? why would this be undefined, and is defaulting to 1 safe? do we not want to support 721s in this context at all?

Reply:
(by safe I mean, could defaulting to 1 lead to simulating the wrong NFT that we actually want to scrape if for some reason scraping the actual token ID failed?)

Reply:
Zora sometimes uses marketing URLs without the trailing /<tokenId> but the tokenId to mint seems to be 1.

I'm not 100% sure what the deal is here, but will live and learn!